File: 739-daily-temperatures/daily-temperatures.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int>& temperatures) {
        vector<int> answer(temperatures.size(),0);
        stack<int> s;

        for(int i=temperatures.size()-1; i>=0; i--){
            
            while(!s.empty() && temperatures[s.top()] <= temperatures[i]){
                s.pop();
            }
            if(!s.empty()){
                answer[i] = s.top() - i;
            }
            s.push(i);
        }

        return answer;
    }
};

File: 739-daily-temperatures/daily-temperatures.py
class Solution:
    def dailyTemperatures(self, temperatures: List[int]) -> List[int]:
        result = [0]*len(temperatures)
        stack = []
        for i in range(len(temperatures)-1, -1, -1):
            while (stack and temperatures[stack[-1]] <= temperatures[i]):
                stack.pop()

            if stack:
                result[i] = stack[-1] - i

            stack.append(i)
            
        return result